lib/smooth/core/io/spi/SPIDevice.cpp

```diff
@@ -105,6 +106,50 @@ namespace smooth::core::io::spi
         return res == ESP_OK;
     }
 
+    bool SPIDevice::polling_write(spi_transaction_t& transaction)
+    {
+        // Attach ourselves as the user-data
+        transaction.user = this;
+
+        auto res = spi_device_polling_transmit(dev, &transaction);
+
+        if (res != ESP_OK)
+        {
+            Log::error(log_tag, "polling_write() failed");
+        }
+
+        return res == ESP_OK;
+    }
+
+    bool SPIDevice::queue_transaction(spi_transaction_t& transaction)
+    {
+        // Attach ourselves as the user-data
+        transaction.user = this;
```

Comment:
Are there any threading issues doing this? What if a SPI device is destroyed while a transmission is in progress?

Reply:
Updated SPIDevice.cpp